### ISSUE TITLE
autocomplete missing proper aria roles

### ISSUE THREAD

**Issue by eginwong** [angular/material2#6028]
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
Lighthouse test on the autocomplete component should pass.

#### What is the current behavior?

Finding the following infractions on accessibility for autocomplete:
1. [aria-*] attributes do not match their roles.
2. Elements with [role] that require specific children [role]s, are missing.
3. [aria-*] attributes do not have valid values.

#### What are the steps to reproduce?

Run the lighthouse chrome extension on the Angular Material2 examples page for autocomplete:
https://material.angular.io/components/autocomplete/examples

Note the three infractions related to autocomplete.

#### What is the use-case or motivation for changing an existing behavior?
Accessibility should pass when using the autocomplete.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Using whichever the demo site for angular material2 is currently on.

**Comment by eginwong** [comment]
Similar issues with the [select](https://material.angular.io/components/select/examples) component and the [datepicker](https://material.angular.io/components/datepicker/examples) component.

**Comment by stefan-richardson** [comment]
Any update on this issue?

**Comment by crisbeto** [comment]
These seem to be sorted out in master when running against aXe (which is used internally by Lighthouse). Note that Lighthouse reports some of these as issues, because it refreshes the page before analyzing it, which means that the autocomplete panel isn't rendered in the DOM (we only render it while the input is focused).
